### PR TITLE
Add solution to usaco-693

### PR DESCRIPTION
**Pull request by topazand**

### CODE REVIEW

File: solutions/USACO jan17-balanced-photo.mdx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+An alternative solution uses order statistic tree.(C++ only)
+
+## Solution(C++ only)
+
+In order to find the cows taller than the current cow on its left and right, we maintain two balanced binary search trees storing the heights of the cows on the left and right of the current cow. Then, using function `lower_bound`, we can find the first cow in the set that is higher than the current cow. 
+
+However, STL `set` is **NOT** a random access data-structure, which means that the time complexity of finding all cows higher than the current cow is $O(N)$.(e.g using `distance()`)
+
+To lower the time complexity, we can use an [order statistic tree](https://usaco.guide/gold/PURS/#order-statistic-tree) to find the number of cows that are higher than the current one in $O(Nlog(N))$ time.
```
Comment (dolphingarlic) [inline solutions/USACO jan17-balanced-photo.mdx:20]: Use $\mathcal O$ instead of just $O$

Comment (dolphingarlic) [reply solutions/USACO jan17-balanced-photo.mdx:20]: Also use $\log$ instead of $log$

File: solutions/USACO jan17-balanced-photo.mdx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+author: Andy Wang
+---
+
+[Official Analysis](http://usaco.org/current/data/sol_bphoto_gold_jan17.html)
+
+The official solution uses [binary indexed tree](https://usaco.guide/gold/PURS/#binary-indexed-tree)
+
+An alternative solution uses order statistic tree.(C++ only)
+
+## Solution(C++ only)
```
Comment (dolphingarlic) [inline solutions/USACO jan17-balanced-photo.mdx:14]: Put spaces between brackets and the preceding words

File: solutions/USACO jan17-balanced-photo.mdx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+---
+id: usaco-693
+source: USACO Gold 2017 January
+title: Balanced Photo
+author: Andy Wang
+---
+
+[Official Analysis](http://usaco.org/current/data/sol_bphoto_gold_jan17.html)
+
+The official solution uses [binary indexed tree](https://usaco.guide/gold/PURS/#binary-indexed-tree)
```
Comment (dolphingarlic) [inline solutions/USACO jan17-balanced-photo.mdx:10]:
```suggestion
The official solution uses a [binary indexed tree](https://usaco.guide/gold/PURS/#binary-indexed-tree).
```

File: solutions/USACO jan17-balanced-photo.mdx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+An alternative solution uses order statistic tree.(C++ only)
```
Comment (dolphingarlic) [inline solutions/USACO jan17-balanced-photo.mdx:12]:
```suggestion
An alternative solution uses an order statistic tree.(C++ only)
```

File: solutions/USACO jan17-balanced-photo.mdx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+## Solution(C++ only)
+
+In order to find the cows taller than the current cow on its left and right, we maintain two balanced binary search trees storing the heights of the cows on the left and right of the current cow. Then, using function `lower_bound`, we can find the first cow in the set that is higher than the current cow.
```
Comment (dolphingarlic) [inline solutions/USACO jan17-balanced-photo.mdx:16]:
```suggestion
In order to find the cows taller than the current cow on its left and right, we maintain two balanced binary search trees storing the heights of the cows on the left and right of the current cow. Then, using binary search, we can find the first cow in the set that is higher than the current cow. 
```

File: solutions/USACO jan17-balanced-photo.mdx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+The official solution uses [binary indexed tree](https://usaco.guide/gold/PURS/#binary-indexed-tree)
+
+An alternative solution uses order statistic tree.(C++ only)
+
+## Solution(C++ only)
+
+In order to find the cows taller than the current cow on its left and right, we maintain two balanced binary search trees storing the heights of the cows on the left and right of the current cow. Then, using function `lower_bound`, we can find the first cow in the set that is higher than the current cow. 
+
+However, STL `set` is **NOT** a random access data-structure, which means that the time complexity of finding all cows higher than the current cow is $O(N)$.(e.g using `distance()`)
```
Comment (dolphingarlic) [inline solutions/USACO jan17-balanced-photo.mdx:18]:
```suggestion
However, `std::set` is **NOT** a random access data-structure, which means that the time complexity of finding all cows higher than the current cow is $O(N)$.(e.g using `distance()`)
```

**Review by dolphingarlic — CHANGES_REQUESTED** [review]
There are quite a few grammar errors e.g. not using articles/using the wrong prepositions. You should probably run this through Grammarly or something

File: solutions/USACO jan17-balanced-photo.mdx (hunk 6 above)
Comment (dongliu0426) [reply solutions/USACO jan17-balanced-photo.mdx:18]: also, put the period after the parentheses.

File: solutions/USACO jan17-balanced-photo.mdx (hunk 2 above)
Comment (dongliu0426) [reply solutions/USACO jan17-balanced-photo.mdx:14]: put the period after the parentheses.

File: solutions/USACO jan17-balanced-photo.mdx (hunk 4 above)
Comment (dongliu0426) [reply solutions/USACO jan17-balanced-photo.mdx:12]: put the period after the parentheses.

File: solutions/USACO jan17-balanced-photo.mdx (hunk 3 above)
Comment (dongliu0426) [reply solutions/USACO jan17-balanced-photo.mdx:10]: change link to `/gold/PURS/#binary-indexed-tree`.

File: solutions/USACO jan17-balanced-photo.mdx (hunk 1 above)
Comment (dongliu0426) [reply solutions/USACO jan17-balanced-photo.mdx:20]: change link to `/gold/PURS/#order-statistic-tree`.

**Review by darren-yao — APPROVED** [review]

**Comment by topazand** [comment]
thanks for fixing that... I was preparing for AMC so I didn't have time to check github...sorry

**Pull request merged**